modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/AttributionDocumentGeneratorImpl.java

```diff
@@ -72,15 +77,41 @@ public AttributionDocumentGeneratorImpl(String documentName, File workingDir, St
      * @return (non - null) the file handle of the generated attribution document.
      */
     public File generate(List<ArtifactAndLicense> artifacts) {
-        Templates templates = TemplateLoaderUtil.load(templateKey);
-
-        File title = writeTitle(templates);
-        File copyright = writeCopyright(templates);
-        File backPage = writeBackPage(templates.getBackPage());
-        File artifactPages = writeArtifacts(templates, artifacts, templates.getContent());
-        File intermediateDoc = mergePages(title, copyright, artifactPages, backPage);
-
-        return postProcess(templates, intermediateDoc, documentName);
+        Templates templates = null;
+        try {
+            templates = TemplateLoaderUtil.load(templateKey);
```

Comment:
If you would add a close method to Templates class and let it implement Closable, you could ease this here with try (Templates templates = ...) and move the close complexity to Templates.